src/main/java/com/bobocode/petros/container/ApplicationAnnotationContainer.java

```diff
@@ -17,12 +18,30 @@ public ApplicationAnnotationContainer(String packageName) {
 
     @Override
     public <T> T getDependency(String name, Class<T> clazz) {
-
-        return null;
+        return (T) dependencyMap.values().stream()
+                .filter(obj -> obj.getClass().getName().toLowerCase().equals(name.toLowerCase())
```

Comment:
It's better to search Dependency object by DependencyDefinition _name_ field, since, for example, _configClass_ dependencies are using method names as a name of Dependency

Reply:
additional , on my opinion we can extract lambda in separate method but I don't have a strong opinion regarding this.

Reply:
> It's better to search Dependency object by DependencyDefinition _name_ field, since, for example, _configClass_ dependencies are using method names as a name of Dependency

Dependency object is the value of the map, isn't it?